### PR TITLE
Revise for self-hosting at commit 2b5d7b6

### PR DESCRIPTION
**Pull request by vacantron**
To process the ~700 lines inline libc, the below macro is set:

```diff
+ #define MAX_ANALYSIS_STACK_SIZE 700
```

Now, `shecc` can bootstrap itself at the older (2b5d7b6) version.

### CODE REVIEW

File: src/arm-codegen.c [hunk 1]
```diff
@@ -27,18 +27,39 @@ void cfg_flatten()
...
+                elf_offset += 4;
+            break;
         default:
             printf("Unknown opcode\n");
             abort();
         }
     }
-    /* jump to main */
-    elf_offset += 4;
+    /* prepare argc, argv and jump to main */
```
Comment (jserv) [inline src/arm-codegen.c:61]: Refine the grammar.

File: src/arm-codegen.c [hunk 2]
```diff
@@ -226,23 +247,51 @@ void code_generate()
...
+            } else
+                emit(__sw(__AL, rn, __r12, ph2_ir->src1));
             break;
         default:
             printf("Unknown opcode\n");
             abort();
         }
     }
-    /* jump to main */
+    /* prepare argc, argv and jump to main */
```
Comment (jserv) [inline src/arm-codegen.c:289]: Ditto.

File: src/defs.h [hunk 3]
```diff
@@ -37,6 +37,7 @@
 #define MAX_CASES 128
 #define MAX_NESTING 128
 #define MAX_OPERAND_STACK_SIZE 32
+#define MAX_ANALYSIS_STACK_SIZE 700
```
Comment (jserv) [inline src/defs.h:40]: How can you measure and/or calculate the value?

Comment (vacantron) [reply src/defs.h:40]: Loading an address from .data section increments the stack index of `live_kill` by 1 in analysis. The inline libc has ~700 times loading, so the limitation is set to 700.

File: src/arm-codegen.c (hunk 1 above)
Comment (vacantron) [reply src/arm-codegen.c:61]: They are "prepare argc, argv" and "jump to main". Are they too confusing?

File: src/defs.h (hunk 3 above)
Comment (jserv) [reply src/defs.h:40]: You should explain within git commit message(s).

File: src/arm-codegen.c (hunk 1 above)
Comment (jserv) [reply src/arm-codegen.c:61]: Yes, use Grammarly or similar tools to refine.

File: src/arm-codegen.c [hunk 4]
```diff
@@ -27,18 +27,39 @@ void cfg_flatten()
          ph2_ir = ph2_ir->next) {
         switch (ph2_ir->op) {
         case OP_load_constant:
+            if (ph2_ir->src0 < 0)
```
Comment (jserv) [inline src/arm-codegen.c:30]: Shall we introduce a helper function for the adjust of `elf_offset`?

File: src/riscv-codegen.c [hunk 5]
```diff
@@ -54,107 +119,259 @@ void cfg_flatten()
...
         }
     }
 }
 
+void emit(int code)
+{
+    elf_write_code_int(code);
+}
+
+void emit_from_ph2_ir(ph2_ir_t *ph2_ir)
```
Comment (jserv) [inline src/riscv-codegen.c:147]: Use shorter form. e.g., `emit_ph2ir`.

**Comment by jserv** [comment]
Thank @vacantron for contributing!

**Pull request merged**